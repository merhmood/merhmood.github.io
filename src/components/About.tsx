import React from 'react'

const About = () => {
  return (
    <div className='about'>
        <div>
            <ul>
                <li></li>
                <li></li>
            </ul>
        </div>
    </div>
  )
}

export default About