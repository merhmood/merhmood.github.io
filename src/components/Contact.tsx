import React from 'react'

const Contact = () => {
  return (
    <div className='contact' id="contact">
      <h3>Contact</h3>
      <div>
        <span>
          <a 
          href="https://www.linkedin.com/in/mahmud-suberu-684020235/"
          >
            LinkedIn
          </a>
        </span>
        <span><a href="https://twitter.com/merh_mood">Twitter</a></span>
        <span><a href="https://www.instagram.com/merh_mood/">Instagram</a></span>
      </div>
    </div>
  )
}

export default Contact