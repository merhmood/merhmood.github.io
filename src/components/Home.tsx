import React, { useEffect, useState } from 'react'

const Home = () => {

  const [profilePic, setProfilePic] = useState('')

  useEffect(()=>{
    setProfilePic(
      `${process.env.PUBLIC_URL}/assets/profile-img.jpg`
    )
  }, [])

  return (
    <div className='home'>
      <div className='my-self'>
        <div className='image-container'>
          <img src={profilePic} alt='profile pic' />
        </div>
        <span>Mahmud Suberu</span>
        <p>
          I'm a software engineer from <span>Nigeria</span>
          , currently using the MERN Stack for my 
          day to day development workflow.
          I have great feel for Software Design, 
          Software Architecture, clean code, 
          Test Driven Development, DevOps and 
          Agile Framework &#x1F601;	

        </p>
      </div>
    </div>
  )
}

export default Home