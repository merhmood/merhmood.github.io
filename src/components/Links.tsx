import {useState, useEffect} from"react"

type ScrollPosition = {
    [index: string]: number
    Skills: number
    Projects: number
    About: number
    Resume: number
}

const scrollObject: ScrollPosition = { Skills: 0, Projects: 0, About: 0, Resume: 0 }

function Links() {
    const [scrollPosition, setScrollPosition] = useState<ScrollPosition>({ ...scrollObject })
    useEffect(() => {
        let App = document.querySelectorAll("section")
        let tempScrollPosition = { ...scrollObject }
        for (let i = 2; i < App.length; i++) {
            let attrName: string = App[i].children[0].innerHTML
            tempScrollPosition[attrName] = App[i].offsetTop

        }
        setScrollPosition(tempScrollPosition as ScrollPosition)
    }, [])
  return (
    <ul>
        <li onClick={() => scrollPage("Skills", scrollPosition)}>Skills</li>
        <li onClick={() => scrollPage("Projects", scrollPosition)}>Projects</li>
        <li onClick={() => scrollPage("About", scrollPosition)}>About</li>
        <li onClick={() => scrollPage("Resume", scrollPosition)}>Resume</li>
    </ul>
  )
}
function scrollPage(link: string, scrollPosition: ScrollPosition) {
    window.scrollTo(0, scrollPosition[link])
}

export default Links