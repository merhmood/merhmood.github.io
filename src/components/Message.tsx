import React from 'react'

function Message() {
  return (
    <div className='message'>
        <h1>
            Welcome to my portfolio website
        </h1>
        <p>-Hi i am mahmud suberu, a Fullstack developer and this is my porfolio website. I hope you have a great time getting to know me</p>
        <button onClick={() => {
            let scrollToPosition = (window.innerHeight / 3) * 2
            window.scrollTo(0, scrollToPosition)
        }}>
            commencer
            <span>&#8595;</span>
        </button>
    </div>
  )
}

export default Message