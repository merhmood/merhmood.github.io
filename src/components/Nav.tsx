import React, { useEffect, useState } from 'react'

type NavLeftSpacingValue = {
  left: number;
} | undefined

type NavProps = {
  navLeftSpacing: ()=> NavLeftSpacingValue
}

const Nav = (props: NavProps) => {
  let [cssStyle, setCssStyle] = useState<NavLeftSpacingValue>(()=>{
    return props.navLeftSpacing()
  })
  useEffect(()=>{
    window.addEventListener(
      'resize', ()=>{
        resizeListener(props, setCssStyle)
    })
    return ()=>{
      window.removeEventListener(
        'resize', ()=>{
          resizeListener(props, setCssStyle)
      })
    }
  }, [props])
  return (
    <nav style={cssStyle}>
        <ul>
        <li><a href="#home">Home</a></li>
        <li><a href="#about">About</a></li>
        <li><a href="#personal-projects">Personal Projects</a></li>
        <li><a href="#contact">Contact</a></li>
        </ul>
    </nav>
  )
}

// This function enable the navigation bar position to
// remain arranged when the user tries to reside the window
// manually

export function resizeListener(propsvalue: NavProps, stateDispatch: React.Dispatch<React.SetStateAction<NavLeftSpacingValue>>) {
  let newCssStyle = propsvalue.navLeftSpacing()
  stateDispatch(newCssStyle)
  return newCssStyle
}


export default Nav