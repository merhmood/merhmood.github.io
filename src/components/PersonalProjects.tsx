import React from 'react'

const PersonalProjects = () => {
  return (
    <div id="personal-projects">PersonalProjects</div>
  )
}

export default PersonalProjects