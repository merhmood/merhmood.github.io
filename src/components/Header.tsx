import React from 'react'

import Nav from './Nav'

const Header = () => {
  return (
    <Nav navLeftSpacing={navLeftSpacing}/>
  )
}
export function navLeftSpacing() {
  const mediaQueries = window.matchMedia('(min-width: 783px)')
  if (mediaQueries.matches) {
    const navWidth = 180
    const appWidth = 660
    const navLeftSpacing = appWidth + ((window.innerWidth - appWidth) / 2) - navWidth
    return { left: navLeftSpacing }
  }
}

export default Header