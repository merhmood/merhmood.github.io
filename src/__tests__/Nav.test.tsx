import React from 'react'
import renderer from 'react-test-renderer'

import { resizeListener } from '../components/Nav';
import Nav from "../components/Nav";


it('testing the nav component width', ()=>{
    const tree = renderer.create(<Nav navLeftSpacing={navLeftSpacingMock}/>)
    expect(tree.toJSON()).toMatchSnapshot()

    function navLeftSpacingMock() {
        return {left: 200}
    }
})
it('testing to check if the resize listener was called correctly',()=>{
    
    const propsStub = {
        navLeftSpacing(){
            return {left: 700}
        }
    }
    const stateDispatchStub = jest.fn()

    let expectValue = resizeListener(propsStub, stateDispatchStub)
    expect(expectValue).toStrictEqual({ left: 700 })
})

it('testing to check if the stateDispatch was called correctly', () => {

    const propsStub = {
        navLeftSpacing() {
            return { left: 300 }
        }
    }

    const stateDispatchMock = jest.fn()

    resizeListener(propsStub, stateDispatchMock)

    const argument = stateDispatchMock.mock.calls[0]
    const calledWithArgument = [{ left: 300 }]
    expect(argument).toEqual(calledWithArgument)
})

it('testing to check if the props navLeftSpacing was called correctly', () => {

    const propsMock = {
        navLeftSpacing: jest.fn(()=> {
            return { left: 300 }      
        })
    }

    const stateDispatchStub = jest.fn()

    resizeListener(propsMock, stateDispatchStub)

    const receivedValue = propsMock.navLeftSpacing.mock.results
    const expectedValue = [{ type: "return", value: { left: 300 } }]
    expect(expectedValue).toEqual(receivedValue)
})