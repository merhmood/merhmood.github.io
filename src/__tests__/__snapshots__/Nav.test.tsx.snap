// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the nav component width 1`] = `
<nav
  style={
    Object {
      "left": 200,
    }
  }
>
  <ul>
    <li>
      <a
        href="#home"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="#about"
      >
        About
      </a>
    </li>
    <li>
      <a
        href="#personal-projects"
      >
        Personal Projects
      </a>
    </li>
    <li>
      <a
        href="#contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
